src/main/java/com/fasterxml/jackson/dataformat/xml/util/DefaultXmlPrettyPrinter.java

```diff
@@ -79,10 +79,24 @@
 
     /**
      * By default, will try to set as System.getProperty("line.separator")
-     * can later set custom lineFeed with withCustomlineFeed method
+     * Can later set custom lineFeed with withCustomlineFeed method
      * @since 2.15
      */
-    protected static String _lineFeed;
+    private final static String DEFAULT_LINE_FEED;
+    static {
+        String lf = null;
+        try {
+            lf = System.getProperty("line.separator");
+        } catch (Throwable t) { } // access exception?
+
+        if (lf != null) {
+            DEFAULT_LINE_FEED = lf;
+        } else {
+            DEFAULT_LINE_FEED = "\n"; // incase system has changed property name? line.separator
+        }
+    }
+    
+    protected static String _lineFeed = DEFAULT_LINE_FEED;
```

Comment:
`protected String _lineFeed = DEFAULT_LINE_FEED;` -- should not be static

Reply:
@pjfanning I have a question. 
Even if we make `_lineFeed` field non-static, we have static line-feeding Indenters like `Lf2SpacesIndenter`. Linefeed happens inside `Lf2SpacesIndenter.writeIndentation()`.
So Indenters can't access non-static field of outer class. 🥲

How can we solve this?

Reply:
My thoughts, line-feed outside of Lf2SpacesIndenter?

Reply:
Lf2SpacesIndenter could have a constructor that takes an instance of DefaultXmlPrettyPrinter in its constructor. Or we could make Lf2SpacesIndenter non-static.